### DISCUSSION TITLE
Nextjs axios request to sanctum api works on frontend but not on the backend {middleware and api routes}

### DISCUSSION THREAD

**Discussion by yassirelkhaili** [community/community#110201] [Programming Help]
### Body

Hi, 
Context: I am working on this basic authentication system using Nextjs and laravel sanctum, I have the user registration setup correctly and I can register and logout everything works fine except when I make a request in the middleware to the /api/user endpoint to retrieve the user, then I get error 401 Unauthorized even when I have logged in on the frontend, the weird thing is that the same request works when I make it inside a useEffect in the frontend and returms 200 ok with the user info in the response. my frontend is hosted on localhost:3000 whilst my backend is on localhost:8000
env sanctum config:
<img width="519" alt="Capture" src="https://github.com/community/community/assets/60744484/da2fdbf2-32e5-4af7-92a2-821095c61229">
sanctum api routes:
<img width="530" alt="Capture2" src="https://github.com/community/community/assets/60744484/703b10e9-298c-411b-9dbd-ed55c14013c1">
nextjs frontend (this returns ok 200 response with user info):
<img width="603" alt="Captureok" src="https://github.com/community/community/assets/60744484/ca6d003e-dde4-4e06-a79a-7f78f900b6bd">
nextjs api route (this returns 401 Unauthorized):
<img width="495" alt="Capture23" src="https://github.com/community/community/assets/60744484/92415d28-ff96-4caf-8094-8e5b16122653">
objective: to make an http request to sanctum api to see if user is authenticated or not then send him either to the dashboard or login page (this must be done either inside the middleware or the backend) I appreciate any help I could get to solve this issue

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by wgarciar3** [comment]
The difference is where each request actually runs. useEffect executes in the browser, so the browser automatically attaches your session/XSRF cookies to the axios call (that's why it returns 200). Next.js middleware runs server-side (Edge/Node runtime) — it's essentially a separate HTTP client with no access to the browser's cookie jar, so axios sends the request with no cookies at all unless you forward them explicitly.

Fix: read the cookies off the incoming request and attach them manually to the outgoing call:


import { NextResponse } from 'next/server';
import axios from 'axios';

export async function middleware(request) {
  const cookie = request.headers.get('cookie') ?? '';

  try {
    await axios.get('http://localhost:8000/api/user', {
      headers: { Cookie: cookie },
      withCredentials: true,
    });
    return NextResponse.next();
  } catch (e) {
    return NextResponse.redirect(new URL('/login', request.url));
  }
}
Also double check SANCTUM_STATEFUL_DOMAINS=localhost:3000 is set in your Laravel .env (needed since frontend/backend are on different ports) — that part is clearly already working since the useEffect call succeeds, but worth confirming if you still see issues after forwarding the cookie.
